priv/templates/phx.gen.live/index.ex

```diff
@@ -0,0 +1,43 @@
+defmodule <%= inspect context.web_module %>.<%= inspect Module.concat(schema.web_namespace, schema.alias) %>Live.Index do
+  use <%= inspect context.web_module %>, :live_view
+
+  alias <%= inspect context.module %>
+  alias <%= inspect schema.module %>
+
+  def render(assigns) do
+    Phoenix.View.render(<%= inspect context.web_module %>.<%= inspect Module.concat(schema.web_namespace, schema.alias) %>View, "index.html", assigns)
+  end
+
+  def mount(_params, _session, socket) do
+    {:ok,
+     socket
+     |> assign(page_title: gettext("Listing <%= schema.human_plural %>"))
+     |> assign(<%= schema.singular %>: nil)}
+  end
+
+  def handle_params(params, _, socket) do
+    <%= schema.singular %> =
+      case socket.assigns.live_view_action do
+        :edit -> <%= inspect context.alias %>.get_<%= schema.singular %>!(params["id"])
+        :new -> %<%= inspect schema.alias %>{}
+        :index -> nil
+      end
+
+    {:noreply, fetch(assign(socket, <%= schema.singular %>: <%= schema.singular %>))}
```

Comment:
Calling `fetch` here would reload all entries when we close the modal. Perhaps this is not what we want right now... but I guess users would want this if they add any sorting or pagination. Should we optimize for which scenario?

Reply:
we necessarily have to call fetch if we want the updated entries or newly added entry to appear. The other option is going the fully optimized route of temporary_assigns + phx-update=append, but we were concerned before that it was too much to start with. Still same thoughts?

Reply:
My concern is the following:

1. When we open up the model, we do a query, and that is not necessary

2. When we close the model, we reload the data, and that is not necessary either

3. This may lead to usability issues, as people may not expect the underlying list to reload when I cancel the model.

However, I think we can solve 1 and 2 by rewriting it as:

```elixir
      case socket.assigns.live_view_action do
        :edit ->
          {:noreply, assign(socket, post: <%= inspect context.alias %>.get_<%= schema.singular %>!(params["id"]))}

        :new ->
          {:noreply, assign(socket, post: %<%= inspect schema.alias %>{})}

        :index ->
          {:noreply, assign(socket, post: nil) |> fetch()}
      end
```

Maybe that makes it good enough? Thoughts?